File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/component/gift/view/animation/AnimationView.java
package com.trtc.uikit.livekit.component.gift.view.animation;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public abstract class AnimationView extends FrameLayout {
    protected Callback mCallback;

    public AnimationView(@NonNull Context context) {
        this(context, null);
    }

    public AnimationView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public abstract void playAnimation(String playUrl);

    public void setCallback(Callback callback) {
        mCallback = callback;
    }

    public interface Callback {
        void onFinished(int error);
    }
}

File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/component/gift/view/animation/AnimationView.kt
package com.trtc.uikit.livekit.component.gift.view.animation

import android.content.Context
import android.util.AttributeSet
import android.widget.FrameLayout

abstract class AnimationView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) :
    FrameLayout(context, attrs) {
    var callback: Callback? = null
    var roomId: String = ""
    abstract fun playAnimation(playUrl: String)
    abstract fun stopPlay()
    interface Callback {
        fun onFinished(error: Int)
    }
}